docs/src/chainrules.md

```diff
@@ -0,0 +1,243 @@
+# ChainRules integration
```

Comment:
shouldnt we do this with a tiny toy example with 2 variables and 2 constraints max?
The whole concept of UC might push people away.
Then we do a UC version in a tutorial.

Reply:
Yes I agree, I think we can add another one that's simpler? This one is still interesting because non-trivial

Reply:
sure we can have both!
Once we have the simple one, I would put the simple in the manual and this complex one in the tutorials.

Reply:
tutorials still in these docs right?